codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGenerator.kt

```diff
@@ -524,8 +527,12 @@ class ServerBuilderGenerator(
                             }
                         }
                     }
-                    serverBuilderConstraintViolations.forMember(member)?.also {
-                        rust(".ok_or(ConstraintViolation::${it.name()})?")
+                    if (member.hasNonNullDefault()) {
```

Comment:
We need to inject the writable _before_ we check constraints, so that default values undergo validation too.

At least until Smithy validates the constraints and default values at build time.